crates/corrosion/src/command/consul/sync.rs

```diff
@@ -195,6 +196,123 @@ async fn setup(
     Ok(())
 }
 
+async fn update_hashes(
+    corrosion: &CorrosionClient,
+    nodename: &str,
+    service_hashes: &mut HashMap<String, u64>,
+    check_hashes: &mut HashMap<String, u64>,
+) -> eyre::Result<()> {
+    let (services, checks) = {
+        let conn = corrosion.pool().get().await?;
+
+        let services = conn
+            .prepare(
+                "SELECT id, name, tags, meta, port, address FROM consul_services WHERE node = ?",
+            )?
+            .query_map([nodename], |row| {
+                let tag_json: String = row.get(2)?;
+                let meta_json: String = row.get(3)?;
+                Ok(AgentService {
+                    id: row.get(0)?,
+                    name: row.get(1)?,
+                    tags: serde_json::from_str(&tag_json).map_err(|e| {
+                        rusqlite::Error::FromSqlConversionFailure(
+                            0,
+                            rusqlite::types::Type::Text,
+                            Box::new(e),
+                        )
+                    })?,
+                    meta: serde_json::from_str(&meta_json).map_err(|e| {
+                        rusqlite::Error::FromSqlConversionFailure(
+                            0,
+                            rusqlite::types::Type::Text,
+                            Box::new(e),
+                        )
+                    })?,
+                    port: row.get(4)?,
+                    address: row.get(5)?,
+                })
+            })
+            .map_err(|e| eyre::eyre!("could not query consul_checks' table_info: {e}"))?
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let checks = conn.prepare("SELECT id, name, status, output, service_id, service_name FROM consul_checks WHERE node = ?")?
+            .query_map([nodename], |row| {
+                Ok(AgentCheck{
+                    id: row.get(0)?,
+                    name: row.get(1)?,
+                    status: row.get(2)?,
+                    output: row.get(3)?,
+                    service_id: row.get(4)?,
+                    service_name: row.get(5)?,
+                    notes: None,
+                })
+            })
+            .map_err(|e| eyre::eyre!("could not query consul_checks' table_info: {e}"))?
+            .collect::<Result<Vec<_>, _>>()?;
+
+        (services, checks)
+    };
+
+    let mut statements = Vec::with_capacity(services.len() + checks.len());
+
+    let mut svc_insert = vec![];
+    let mut checks_insert = vec![];
+
+    for svc in services {
+        if service_hashes.get(&svc.id).is_none() {
+            let hash = hash_service(&svc);
+            statements.push(Statement::WithParams(
+                "INSERT INTO __corro_consul_services ( id, hash )
+    VALUES (?, ?)
+    ON CONFLICT (id) DO UPDATE SET
+        hash = excluded.hash;"
```

Comment:
Probably easier to use `INSERT OR REPLACE INTO` instead of a query with an `ON CONFLICT`, given we always want to replace.